### PR TITLE
WIP #836 Disable symbolic link removal on update

### PR DESCRIPTION
**Pull request by muuki88**
Puts the symlink creation into the `%intsall` section as mentioned in #836 .

To test: does relocation still works?

### CODE REVIEW

**Comment by jneko33** [comment]
This looks good, will help a lot if it could be merged soon (after the builds pass of course)

**Comment by muuki88** [comment]
So the resulting `install` section looks something like this

```bash
%install
if [ -e "$RPM_BUILD_ROOT" ];
then
  mv "/home/dev/sbt-native-packager/test-project-simple/target/rpm/tmp-buildroot"/* "$RPM_BUILD_ROOT"
else
  mv "/home/dev/sbt-native-packager/test-project-simple/target/rpm/tmp-buildroot" "$RPM_BUILD_ROOT"
fi


relocateLink() {
  if [ -n "$4" ] ;
  then
    RELOCATED_INSTALL_DIR="$4/$3"
    echo "${1/$2/$RELOCATED_INSTALL_DIR}"
  else
    echo "$1"
  fi
}
rm -rf $(relocateLink /usr/bin/test-project-simple /usr/share/test-project-simple test-project-simple $RPM_INSTALL_PREFIX) && ln -s $(relocateLink /usr/share/test-project-simple/bin/test-project-simple /usr/share/test-project-simple test-project-simple $RPM_INSTALL_PREFIX) $(relocateLink /usr/bin/test-project-simple /usr/share/test-project-simple test-project-simple $RPM_INSTALL_PREFIX)
rm -rf $(relocateLink /etc/test-project-simple /usr/share/test-project-simple test-project-simple $RPM_INSTALL_PREFIX) && ln -s $(relocateLink /usr/share/test-project-simple/conf /usr/share/test-project-simple test-project-simple $RPM_INSTALL_PREFIX) $(relocateLink /etc/test-project-simple /usr/share/test-project-simple test-project-simple $RPM_INSTALL_PREFIX)
rm -rf $(relocateLink /usr/share/test-project-simple/logs /usr/share/test-project-simple test-project-simple $RPM_INSTALL_PREFIX) && ln -s $(relocateLink /var/log/test-project-simple /usr/share/test-project-simple test-project-simple $RPM_INSTALL_PREFIX) $(relocateLink /usr/share/test-project-simple/logs /usr/share/test-project-simple test-project-simple $RPM_INSTALL_PREFIX)
```

Which then fails building with this error message

```
ln: failed to create symbolic link '/usr/bin/test-project-simple': Permission denied
```

I guess this is due to the fact that these symlinks must be relative.

File: src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala [hunk 1]
```diff
@@ -267,17 +264,15 @@ case class RpmSpec(
...
 
     // write build as moving everything into RPM directory.
-    sb append installSection(tmpRoot)
+    sb append installSection(tmpRoot, buildSymlinkScript(meta.name, installDir, symlinks))
     // TODO - Allow symlinks
 
     // write scriptlets
     sb append scriptlets.pretransContent()
     sb append scriptlets.preContent()
-    sb append scriptlets.postContent(buildSymlinkScript(meta.name, installDir, symlinks))
```
Comment (jneko33) [inline src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala]: Was the whole post scriptlet supposed to be removed?

File: src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala [hunk 2]
```diff
@@ -267,17 +264,15 @@ case class RpmSpec(
...
     sb append scriptlets.verifyscriptContent()
     sb append scriptlets.posttransContent()
     sb append scriptlets.preunContent()
-    sb append scriptlets.postunContent(teardownSymlinkScript(meta.name, installDir, symlinks))
```
Comment (jneko33) [inline src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala]: Same with the post uninstall scriptlet?

File: src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala (hunk 1 above)
Comment (muuki88) [inline src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala]: Actually yes. If I understood the rpm docs correctly, this should be handled by the `%install` section.

**Comment by muuki88** [comment]
this pr may get closed in favor of an implementation from @dpennell

**Pull request closed without merging**